src/createHistory.js

```diff
@@ -0,0 +1,22 @@
+import { createStore } from 'redux';
+
+import Actions from './Actions';
+import ActionTypes from './ActionTypes';
+import createHistoryEnhancer from './createHistoryEnhancer';
+import createStoreHistory from './createStoreHistory';
+
+function reducer(state, action) {
+  if (action.type === ActionTypes.LOCATION_UPDATED) {
+    return action.payload;
+  }
+
+  return state;
+}
+
+export default function createHistory(protocol, ...middlewares) {
+  const enhancer = createHistoryEnhancer(protocol, ...middlewares);
+  const store = createStore(reducer, enhancer);
+  store.dispatch(Actions.init());
+
+  return createStoreHistory(store, state => state);
```

Comment:
my redux is rusty, so i'm finidng the naming here confusing, its a bunch of things that have very similar names. the `enhancer` is applying any history specific middlewares (plus your own)? 

Why `createStoreHistory` and the enhancer? To avoid adding methods to the redux Store?